### DISCUSSION TITLE
Unable to access req.user in _app.js when Custom Server sets req.user = userId

### DISCUSSION THREAD

**Discussion by taylor-lindores-reeves** [vercel/next.js#15970] [Help]
I'm using NextJS, Prisma, GraphQL-Yoga, Apollo Client. Currently in my custom server I have the following:

```
server.express.use(async (req, res, next) => {
  if (!req.userId) return next();

  const user = await server
    .context()
    .db.query.user({ where: { id: req.userId } }, "{ id, permissions, email }");

  req.user = user;
  next();
});
```

I would like to access the req.user client-side i.e. in my _app.js file I should be able to do this:

```
class MyApp extends App {
  static async getInitialProps({ Component, ctx }) {
    let pageProps = {};
    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps(ctx);
    }
    // this exposes the query to the user
    pageProps.user = ctx.req.user;
    console.log(ctx.req.userId);
    return { pageProps };
  }

render() {
    const { Component, apollo, pageProps } = this.props;
    return (
           <Component {...pageProps} />
    );
}

```

For some reason req.user is undefined. Any ideas why? I have another application with a similar stack and I have this working. However, this app it doesn't seem to work. Any help greatly appreciated.

**Comment by KB1RMA** [comment]
What does the rest of your custom server look like? You'll need middleware passing the request to Next _after_ your example middleware retrieving a custom user.

```js
const app = next()
const handle = app.getRequestHandler()

// other app prep

// example middleware

server.all('*', (req, res) => {
    return handle(req, res)
})
```

Also - If you're doing client-side routing, your example `getInitialProps` is going to throw as `ctx.req` will be undefined
